src/app/backend/replicasetpodsmetrics.go

```diff
@@ -0,0 +1,147 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"log"
+
+	"encoding/json"
+	"fmt"
+	heapster "k8s.io/heapster/api/v1/types"
+	"k8s.io/kubernetes/pkg/api"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+	"strings"
+)
+
+const (
+	cpuUsage                 = "cpu-usage"
+	memoryUsage              = "memory-usage"
+	megaByte                 = 1024 * 1024
+	defaultHeapsterNamespace = "kube-system"
+	defaultHeapsterScheme    = "http"
+	defaultHeapsterService   = "heapster"
+	defaultHeapsterPort      = "" // use the first exposed port on the service
+)
+
+// Metrics map by pod name.
+type ReplicaSetMetricsByPod struct {
+	// Metrics map by pod name
+	MetricsMap map[string]PodMetrics `json:"metricsMap"`
+}
+
+// Pod metrics structure.
+type PodMetrics struct {
+	// Cumulative CPU usage on all cores in nanoseconds.
+	CpuUsage uint64 `json:"cpuUsage"`
+	// Pod memory usage in MB.
+	MemoryUsage uint64 `json:"memoryUsage"`
+}
+
+// Return Pods metrics for Replica Set or error when occurred.
+func getReplicaSetPodsMetrics(podList *api.PodList, client *client.Client,
+	heapsterClient *client.Client, namespace string, replicaSet string) (
+	*ReplicaSetMetricsByPod, error) {
+	log.Printf("Getting Pods metrics for Replica Set %s in %s namespace",
+		replicaSet,
+		namespace)
+	podNames := make([]string, 0)
+
+	pods := podList.Items
+	for _, pod := range pods {
+		podNames = append(podNames, pod.Name)
+	}
+
+	metricCpuUsagePath := createMetricPath(namespace, podNames, cpuUsage)
+	metricMemUsagePath := createMetricPath(namespace, podNames, memoryUsage)
+
+	resultCpuUsageRaw, err := getRawMetrics(client, heapsterClient, metricCpuUsagePath)
+	if err != nil {
+		return nil, err
+	}
+
+	resultMemUsageRaw, err := getRawMetrics(client, heapsterClient, metricMemUsagePath)
+	if err != nil {
+		return nil, err
+	}
+
+	cpuMetricResult, err := unmarshalMetrics(resultCpuUsageRaw)
+	if err != nil {
+		return nil, err
+	}
+	memMetricResult, err := unmarshalMetrics(resultMemUsageRaw)
+	if err != nil {
+		return nil, err
+	}
+	return createResponse(cpuMetricResult, memMetricResult, podNames), nil
+}
+
+// Create URL path for metrics.
+func createMetricPath(namespace string, podNames []string, metricName string) string {
+	return fmt.Sprintf("/api/v1/model/namespaces/%s/pod-list/%s/metrics/%s",
+		namespace,
+		strings.Join(podNames, ","),
+		metricName)
+}
+
+// Retrieves raw metrics from Heapster.
+func getRawMetrics(client *client.Client, heapsterClient *client.Client, metricPath string) ([]byte, error) {
+
+	var resultRaw []byte
+	var err error
+        // Service Proxy is used when heapster client is nil. Heapster probably installed inside cluster.
+	if heapsterClient == nil {
```

Comment:
You should not be doing this logic here. Just create a RESTClient in dashboard.go (or heapsterclient.go) and use it here.